addons/hr_work_entry_contract/models/hr_work_entry.py

```diff
@@ -73,19 +73,39 @@ def _compute_date_stop(self):
                 continue
             super(HrWorkEntry, work_entry)._compute_date_stop()
 
-    def _get_duration(self, date_start, date_stop):
-        if not date_start or not date_stop:
-            return 0
-        if self._get_duration_is_valid():
-            calendar = self.contract_id.resource_calendar_id
-            if not calendar:
-                return 0
-            employee = self.contract_id.employee_id
-            contract_data = employee._get_work_days_data_batch(
-                date_start, date_stop, compute_leaves=False, calendar=calendar
-            )[employee.id]
-            return contract_data.get('hours', 0)
-        return super()._get_duration(date_start, date_stop)
+    def _get_duration_compute_from_calendar(self):
+        self.ensure_one()
+        return not self._get_duration_is_valid()
```

Comment:
```suggestion
    def _is_duration_computed_from_calendar(self):
        self.ensure_one()
        return not self._get_duration_is_valid()
```
I find it weird for a `_get_duration` method to return a bool :thinking: